jdaviz/core/template_mixin.py

```diff
@@ -491,13 +492,16 @@ def _on_layers_changed(self, msg=None):
         viewers = [self._get_viewer(viewer) for viewer in viewer_names]
 
         manual_items = [{'label': label} for label in self.manual_options]
-        layers = [layer for viewer in viewers for layer in viewer.layers]
-        # remove duplicates - NOTE: by doing this, any color-mismatch between layers with the
-        # same name in different viewers will be randomly assigned within plot_options
-        # based on which was found _first.
-        layer_labels = [layer.layer.label for layer in layers]
-        _, inds = np.unique(layer_labels, return_index=True)
-        layers = [layers[i] for i in inds]
+        if viewers == [None]:
```

Comment:
all these extra checks to handle no viewers concern me a bit... I'd expect `viewers` to be `[]` not `[None]`.  Can either of the list-comprehensions above be edited to account for the `None` case instead of the extra if-statement?  Or is there somewhere more general in the mixins that needs to more gracefully handle the absence of any viewer?

Reply:
I will withhold my review until this concern is addressed.

Reply:
How does this play in with #1681?